src/selectors/functions/InlineFunction.js

```diff
@@ -0,0 +1,72 @@
+import Selector from '../Selector';
+import Specificity from '../Specificity';
+import Sequence from '../dataTypes/Sequence';
+import FunctionValue from '../dataTypes/FunctionValue';
+
+/**
+ * @extends Selector
+ */
+class InlineFunction extends Selector {
+	/**
+	 * @param  {!Array<!Array<string>>}  paramDescriptions  An array of tuples of name and type of the parameters
+	 * @param  {string}                returnType
+	 * @param  {!Selector}              functionBody
+	 */
+	constructor (paramDescriptions, returnType, functionBody) {
+		super(new Specificity({
+			[Specificity.EXTERNAL_KIND]: 1
+		}), Selector.RESULT_ORDERINGS.UNSORTED);
+
+		this._paramDescriptions = paramDescriptions;
+		this._returnType = returnType;
+		this._functionBody = functionBody;
+
+		var executeFunction = (dynamicContext, ...parameters) => {
```

Comment:
check if this is the right dynamicContext - don't you need the one at the point of declaration?